builder/proxmox/artifact.go

```diff
@@ -34,7 +38,7 @@ func (a *Artifact) String() string {
 }
 
 func (a *Artifact) State(name string) interface{} {
-	return nil
+	return a.StateData[name]
```

Comment:
I'm not too well informed about what will go in this map, but I assume the builder itself does not need to care about it at all? The Proxmox builder doesn't currently do much at all with the artifact.

Reply:
The StateData will take the map of "generated_data" from the builder's statebag to the post-processors, allowing post-processors to have access to variables that were set when the builder was running.  

This is where we load the data into the statebag:
https://github.com/hashicorp/packer/pull/8632/files#diff-0510d722beaf01f375fd8c2d964c8143R94
Then, we load the state into de StateData
https://github.com/hashicorp/packer/pull/8632/files#diff-232c9fb4c2936049ef3edc3db021b2c5R121
and this is when the post-processor uses the generated_data:
https://github.com/hashicorp/packer/pull/8632/files#diff-8a3dbac8a2acb14d8a823a2d4651a541R141


It's one way of extending [this feature](https://www.packer.io/docs/templates/engine.html#build) to the post-processors

Reply:
Does it make sense? Please, let me know if it doesn't!

Reply:
I'm just thinking about what a post processor could do in the specific Packer case - I don't think any of the existing post-processors could do much with the "artifact" that is produced (currently it is very dumb, basically just an int referencing the id on the remote system). So just wondering if I'd need to make some changes to make sure this will work?

Reply:
Mostly this is being stored so people can spit it out in the manifest postprocessor if they want to have access to instance details later.